rfcs/0024-nondeterministic-functions.md

```diff
@@ -0,0 +1,60 @@
+---
+feature: non_deterministic_functions
+authors:
+  - "st1page"
+start_date: "2022/11/21"
+---
+
+# nondeterministic-functions
+
+## Summary
+
+We introduce the `MaterializeProject` stream executor to handle non-deterministic function on updatable stream.
```

Comment:
Prefer that Project executor simply has optional state table with columns only corresponding to either UDF or non-deterministic function

Reply:
LGTM